instrumentation/buildtime/mapping-file/plugin/src/main/java/com/splunk/rum/mappingfile/plugin/utils/MappingFileUploader.kt

```diff
@@ -0,0 +1,156 @@
+            return
+        }
+
+        project.logger.debug(
+            "Splunk RUM: Configuration resolved - realm: '$realm', token configured: ${accessToken.isNotBlank()}"
+        )
+
+        val mappingFile = findMappingFile(variant)
+        if (mappingFile == null) {
+            project.logger.error(
+                """
+                Splunk RUM: Mapping file not found for variant '${variant.name}'
+                
+                Searched locations:
+                ${getMappingFileLocations(variant).joinToString("\n") { "  - ${it.absolutePath}" }}
+                
+                This usually means:
+                1. ProGuard/R8 minification is not properly configured
+                2. The build did not complete successfully
+                3. The mapping file is in a custom location
+                                
+                Please verify that minification is enabled and the build completed without errors.
+                If the mapping file is in a custom location, please disable the uploadEnabled and upload it manually.
+                """.trimIndent()
+            )
+            return
+        }
+
+        project.logger.info(
+            "Splunk RUM: Found mapping file (${mappingFile.length()} bytes) at: ${mappingFile.absolutePath}"
+        )
+
+        project.logger.lifecycle("Splunk RUM: Uploading mapping file for variant ${variant.name}")
+        project.logger.info("Splunk RUM: Upload details:")
+        project.logger.info("  File: ${mappingFile.absolutePath}")
+        project.logger.info("  App ID: ${variant.applicationId}")
+        project.logger.info("  Version Code: ${variant.versionCode}")
+        project.logger.info("  Build ID: $buildId")
+        project.logger.debug("  Realm: $realm")
+        project.logger.debug("  File size: ${mappingFile.length()} bytes")
+
+        try {
+            httpClient.uploadMappingFile(
+                mappingFile = mappingFile,
+                buildId = buildId,
+                applicationId = variant.applicationId,
+                versionCode = variant.versionCode,
+                accessToken = accessToken,
+                realm = realm
+            )
+            project.logger.lifecycle("Splunk RUM: Successfully uploaded mapping file")
+        } catch (e: Exception) {
+            project.logger.error("Splunk RUM: Upload failed: ${e.message}")
+            project.logger.debug("Splunk RUM: Full error stacktrace: ${e.stackTraceToString()}")
+            // Don't fail the build, let app build finish
+        }
+    }
+
+    private fun resolveAccessToken(extension: SplunkRumExtension): String {
+        project.logger.debug("Splunk RUM: Resolving access token")
+        return extension.apiAccessToken.orNull
+            ?: project.findProperty("splunk.accessToken") as String?
+            ?: System.getenv("SPLUNK_ACCESS_TOKEN")
+            ?: ""
```

Comment:
Nit: It seems to me that returning null and then checking for it is more standard than using an empty string, but that is a matter of preference.